### PR TITLE
Upgrade freeradius 3.0.21 -> 3.2.3 to address 2 CVEs

### PR DESCRIPTION
**Pull request by arc9693**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Upgrade to 3.2.3 
- Address CVE-2022-41860, CVE-2022-41861
- Update Patch2 & Patch4
- Add Patch6 to address build error
- Add Source105 for user management during installation

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  

- https://nvd.nist.gov/vuln/detail/CVE-2022-41860
- https://nvd.nist.gov/vuln/detail/CVE-2022-41861

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=419788&view=results

### CODE REVIEW

File: SPECS-EXTENDED/freeradius/freeradius.spec [hunk 1]
```diff
@@ -2,8 +2,8 @@ Vendor:         Microsoft Corporation
 Distribution:   Mariner
```
Comment (mfrw) [inline SPECS-EXTENDED/freeradius/freeradius.spec:2]: Optional: If possible give it a partial `lint` :)

File: SPECS-EXTENDED/freeradius/freeradius.spec [hunk 2]
```diff
@@ -2,8 +2,8 @@ Vendor:         Microsoft Corporation
...
 Summary: High-performance and highly configurable free RADIUS server
 Name: freeradius
-Version: 3.0.21
-Release: 9%{?dist}
+Version: 3.2.3
+Release: 1%{?dist}
 License: GPLv2+ and LGPLv2+
 URL: http://www.freeradius.org/
```
Comment (mfrw) [inline SPECS-EXTENDED/freeradius/freeradius.spec:8]: Let's use `https` if it has a TLS enabled website.

```suggestion
URL: https://www.freeradius.org/
```

File: SPECS-EXTENDED/freeradius/freeradius.spec [hunk 3]
```diff
@@ -186,8 +188,9 @@ This plugin provides the REST support for the FreeRADIUS server project.
 %patch1 -p1
 %patch2 -p1
 %patch3 -p1
-%patch4 -p1
+%patch4 -p1 -F 2
 %patch5 -p1
+%patch6 -p1
```
Comment (mfrw) [inline SPECS-EXTENDED/freeradius/freeradius.spec:193]: Lets consider `%autosetup` here.

File: SPECS-EXTENDED/freeradius/freeradius.spec [hunk 4]
```diff
@@ -1,52 +1,45 @@
...
+Source104:      freeradius-tmpfiles.conf
+Source105:      freeradius.sysusers
+Patch1:         freeradius-Adjust-configuration-to-fit-Red-Hat-specifics.patch
+Patch2:         freeradius-Use-system-crypto-policy-by-default.patch
+Patch3:         freeradius-bootstrap-create-only.patch
+Patch4:         freeradius-no-buildtime-cert-gen.patch
+Patch5:         freeradius-bootstrap-make-permissions.patch
+Patch6:         fix-error-for-expansion-of-macro-in-thread.h.patch
 # Is elliptic curve cryptography supported?
 %global HAVE_EC_CRYPTO 1
```
Comment (mfrw) [inline SPECS-EXTENDED/freeradius/freeradius.spec:26]: Let's move this macro as well.

File: SPECS-EXTENDED/freeradius/freeradius.spec [hunk 5]
```diff
@@ -57,6 +50,10 @@ Requires(post): systemd-units
 Requires(post): make
 Requires(preun): systemd-units
 Requires(postun): systemd-units
+%if ! 0%{?rhel}
```
Comment (mfrw) [inline SPECS-EXTENDED/freeradius/freeradius.spec:53]: Lets remove rhel conditional

**Review by mfrw — APPROVED** [review]
I would encourage a build of any dependents of the package as well.

**Comment by arc9693** [comment]
> I would encourage a build of any dependents of the package as well.

No dependents found.

**Review by mfrw — APPROVED** [review]

**Pull request merged**